engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -168,8 +171,10 @@ class WorkflowManagerActor(backend: Backend, config: Config)
 
   when (Running) {
     case Event(SubmitWorkflow(source), _) =>
-      val updatedData = submitWorkflow(source, replyTo = Option(sender), id = None)
-      stay() using updatedData
+      submitWorkflow(source, replyTo = Option(sender), id = None) onSuccess {
+        case updatedData: WorkflowManagerData => self ! UpdateWorkflowManagerState(updatedData)
+      }
+      stay()
```

Comment:
Say the state at the time this code executes is `S`.  Call the workflow being submitted here `A`, so the `UpdateWorkflowManagerState` message is sent to self saying to set the state data to `S + A`.  

But at the time this code is executing, there may already a message to submit workflow `B` in the mailbox.  Because this message is already in the mailbox it will arrive before the `S + A` message.  So this code will fire off an `UpdateWorkflowManagerState` message saying to set the state to `S + B`.  

At this point it's a race to see which of those `UpdateWorkflowManagerState`s arrives first, but the last message to arrive will win and either `A` or `B` will be lost.